File: clients/Ruby-client/lib/solidb/api_keys_client.rb
module SoliDB
  class ApiKeysClient
    def initialize(client)
      @client = client
    end

    def list
      @client.send_command("list_api_keys", {}) || []
    end

    def create(name:, permissions:, expires_at: nil)
      args = { name: name, permissions: permissions }
      args[:expires_at] = expires_at if expires_at
      @client.send_command("create_api_key", args)
    end

    def get(key_id)
      @client.send_command("get_api_key", key_id: key_id)
    end

    def delete(key_id)
      @client.send_command("delete_api_key", key_id: key_id)
      nil
    end

    def regenerate(key_id)
      @client.send_command("regenerate_api_key", key_id: key_id)
    end
  end
end

File: clients/PHP-client/src/ApiKeysClient.php
<?php

namespace SoliDB;

class ApiKeysClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function list(): array
    {
        $res = $this->client->sendCommand('list_api_keys', []);
        return $res['data'] ?? [];
    }

    public function create(string $name, array $permissions, ?string $expiresAt = null): array
    {
        $args = ['name' => $name, 'permissions' => $permissions];
        if ($expiresAt !== null) {
            $args['expires_at'] = $expiresAt;
        }
        $res = $this->client->sendCommand('create_api_key', $args);
        return $res['data'] ?? [];
    }

    public function get(string $keyId): array
    {
        $res = $this->client->sendCommand('get_api_key', ['key_id' => $keyId]);
        return $res['data'] ?? [];
    }

    public function delete(string $keyId): void
    {
        $this->client->sendCommand('delete_api_key', ['key_id' => $keyId]);
    }

    public function regenerate(string $keyId): array
    {
        $res = $this->client->sendCommand('regenerate_api_key', ['key_id' => $keyId]);
        return $res['data'] ?? [];
    }
}
